### ISSUE TITLE
there is a little error in the make_trg_mask function in tutorial 6

### ISSUE THREAD

**Issue by zhangjianzhang** [bentrevett/pytorch-seq2seq#96]
According to the explanation above the code, `trg_pad_mask = (trg != self.trg_pad_idx).unsqueeze(1).unsqueeze(3)` should be changed to `trg_pad_mask = (trg != self.trg_pad_idx).unsqueeze(1).unsqueeze(2)`

lets use a code example to illustrate the error:
suppose `trg_pad_mask` is a batch of 2 sentences tensors, and each of the sentence is of 3 length.

The result of `trg_pad_mask = (trg != self.trg_pad_idx).unsqueeze(1).unsqueeze(2)` is reasonable and consistent with your explanation about target sequence masking above the code.
```python
import torch

trg_pad_mask = torch.tensor([[1,1,1,0,0],[1,1,1,0,0]])

print(trg_pad_mask)

tensor([[1, 1, 1, 0, 0],
        [1, 1, 1, 0, 0]])

trg_pad_mask = trg_pad_mask.unsqueeze(1).unsqueeze(2)

print(trg_pad_mask)

tensor([[[[1, 1, 1, 0, 0]]],
        [[[1, 1, 1, 0, 0]]]])

trg_sub_mask = torch.tril(torch.ones((5,5)))

print(trg_sub_mask)

tensor([[1., 0., 0., 0., 0.],
        [1., 1., 0., 0., 0.],
        [1., 1., 1., 0., 0.],
        [1., 1., 1., 1., 0.],
        [1., 1., 1., 1., 1.]])

trg_mask = trg_pad_mask.byte() & trg_sub_mask.byte()

print(trg_mask)

tensor([[[[1, 0, 0, 0, 0],
          [1, 1, 0, 0, 0],
          [1, 1, 1, 0, 0],
          [1, 1, 1, 0, 0],
          [1, 1, 1, 0, 0]]],
        [[[1, 0, 0, 0, 0],
          [1, 1, 0, 0, 0],
          [1, 1, 1, 0, 0],
          [1, 1, 1, 0, 0],
          [1, 1, 1, 0, 0]]]], dtype=torch.uint8)
```

The result of `trg_pad_mask = (trg != self.trg_pad_idx).unsqueeze(1).unsqueeze(3)` seems to be unreasonable and wrong.
```python
import torch

trg_pad_mask = torch.tensor([[1,1,1,0,0],[1,1,1,0,0]])

print(trg_pad_mask)

tensor([[1, 1, 1, 0, 0],
        [1, 1, 1, 0, 0]])

trg_pad_mask = trg_pad_mask.unsqueeze(1).unsqueeze(3)

print(trg_pad_mask)

tensor([[[[1],
          [1],
          [1],
          [0],
          [0]]],
        [[[1],
          [1],
          [1],
          [0],
          [0]]]])

trg_sub_mask = torch.tril(torch.ones((5,5)))

print(trg_sub_mask)

tensor([[1., 0., 0., 0., 0.],
        [1., 1., 0., 0., 0.],
        [1., 1., 1., 0., 0.],
        [1., 1., 1., 1., 0.],
        [1., 1., 1., 1., 1.]])

trg_mask = trg_pad_mask.byte() & trg_sub_mask.byte()

print(trg_mask)

tensor([[[[1, 0, 0, 0, 0],
          [1, 1, 0, 0, 0],
          [1, 1, 1, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]]],
        [[[1, 0, 0, 0, 0],
          [1, 1, 0, 0, 0],
          [1, 1, 1, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]]]], dtype=torch.uint8)
```
After looking at the Intermediate result, I found that mistake would not influence too much, because in `trg_mask` tensor, the last two zero rows mostly corresponded to pad tokens and the non-zero rows are the same. This mistake maybe cause confusion to the beginners, so I pull this issue.

**Comment by bentrevett** [comment]
Good catch, and thanks for spotting this! 

I'll update this later today.

**Comment by bentrevett** [comment]
I've fixed the error and re-ran the experiments with the fixed unsqueezing. Turns out that it gives the literal exact same results which I think should be the case(?)
